---
	2008-09-30  Felix Riemann  <friemann@svn.gnome.org>
	
	* src/eog-metadata-reader-jpg.c: (eog_metadata_reader_jpg_consume):
	Don't overwrite the last read chunk when skipping over unrecognized
	APP1 chunks. Avoids memory corruption and subsequent segfaults.

Index: trunk/src/eog-metadata-reader-jpg.c
====================================================================
--- trunk/src/eog-metadata-reader-jpg.c	2008/09/30 11:58:28	4810
+++ trunk/src/eog-metadata-reader-jpg.c	2008/09/30 13:49:31	4811
@@ -321,6 +321,7 @@
 			case EJA_OTHER:
 			default:
 				/* skip unknown data */
+				chunk = NULL;
 				priv->state = EMR_SKIP_BYTES;
 				break;
 			}
